torchrl/data/replay_buffers/replay_buffers.py

```diff
@@ -1010,16 +1030,16 @@ def __call__(self, list_of_tds):
         return self.out
 
 
-def _reduce(tensor: torch.Tensor, reduction: str):
+def _reduce(tensor: torch.Tensor, reduction: str, dim: Optional[int] = None):
     """Reduces a tensor given the reduction method."""
     if reduction == "max":
-        return tensor.max().item()
+        return tensor.max().item() if dim is None else tensor.max(dim=dim)[0]
     elif reduction == "min":
-        return tensor.min().item()
+        return tensor.min().item() if dim is None else tensor.min(dim=dim)[0]
     elif reduction == "mean":
-        return tensor.mean().item()
+        return tensor.mean().item() if dim is None else tensor.mean(dim=dim)
     elif reduction == "median":
-        return tensor.median().item()
+        return tensor.median().item() if dim is None else tensor.median(dim=dim)
```

Comment:
what about 
```python
    if reduction == "max":
        result = tensor.max(dim=dim)
    elif reduction == "min":
        result = tensor.min(dim=dim)
    elif reduction == "mean":
        result = tensor.mean(dim=dim)
    elif reduction == "median":
        result = tensor.median(dim=dim)
    else:
        raise NotImplementedError(f"Unknown reduction method {reduction}")
    if isinstance(result, tuple):
        result = result[0]
    return result
```

Reply:
yes that works too, but we also need to call `.item()`